ts/src/main/scala/ch/epfl/ts/traders/Trader.scala

```diff
@@ -22,16 +36,18 @@ abstract class Trader extends Component {
 trait TraderCompanion {
   type Key = String
   
+  def getInstance(uid: Long, parameters: StrategyParameters): Trader
```

Comment:
Why do you want it in the first place? (concrete) Trader is not going to be a singleton, right?